pkg/services/provisioning/values/values.go

```diff
@@ -185,8 +187,29 @@ func transformMap(i map[interface{}]interface{}) interface{} {
 
 // interpolateValue returns final value after interpolation. At the moment only env var interpolation is done
 // here but in the future something like interpolation from file could be also done here.
+// For a literal '$', '$$' can be used to avoid interpolation.
 func interpolateValue(val string) string {
-	return os.ExpandEnv(val)
+	envPlusEscape := make(map[string]string)
+	// Populate environment variable map
+	for _, v := range os.Environ() {
+		sp := strings.SplitN(v, "=", 2) // os.Environ returns key=value pairs, so need to turn it into a map
+		if len(sp) == 1 {
+			sp = append(sp, "")
+		}
+		envPlusEscape[sp[0]] = sp[1]
+	}
+
+	// Add special variable to hold dollar sign
+	escapeKey := "GRAFANA_ESCAPE_DOUBLE_DOLLAR"
+	envPlusEscape[escapeKey] = "$"
+
+	// Turn $$ into a specialEscapeVar for replacement with "$" value
+	escapeVar := fmt.Sprintf("${%v}", escapeKey)
+	val = strings.ReplaceAll(val, "$$", escapeVar)
+
+	return os.Expand(val, func(k string) string {
+		return envPlusEscape[k]
+	})
```

Comment:
Kinda nitpicking here but this seems a lot more code than I would expect. Wouldn't something like:
```
	val = strings.ReplaceAll(val, "$$", "GRAFANA_ESCAPE_DOUBLE_DOLLAR")
	val = os.ExpandEnv(val)
	return strings.ReplaceAll(val, "GRAFANA_ESCAPE_DOUBLE_DOLLAR", "$")
```
also work?
Or to prevent using magic strings that can clash, something like:
```
    parts := strings.Split(val, "$$")
	interpolated := make([]string, len(parts))
	for i, v := range parts {
		interpolated[i] = os.ExpandEnv(v)
	}
	return strings.Join(interpolated, "$")
```